DomainContent/Zaru/applause-app/applause-app.js

```diff
@@ -0,0 +1,420 @@
+        ['right', 'left'].forEach(function(side) {
+
+            for (var i = 0; i < fingerKeys.length; i++) {
+                var finger = fingerKeys[i];
+                var jointSuffixes = 3; // We need to update rotation of the 0, 1 and 2 joints
+                var names = getJointNames(side, finger, jointSuffixes);
+
+                // update every finger joint
+
+                for (var j = 0; j < names.length; j++) {
+                    var index = MyAvatar.getJointIndex(names[j]);
+                    // if no finger is touching restate the default poses
+                    MyAvatar.clearJointData(index);
+                }
+            }
+        });
+    }
+
+    function compareRotations(q1, q2) {
+        var threshold = 0.85;
+        return (Quat.dot(q1, q2) <= threshold);
+    }
+
+    function getRandomApplauseSound() {
+        return INDIVIDUAL_CLAP_URLS[Math.round(Math.random() * INDIVIDUAL_CLAP_URLS.length - 1)];
+    }
+
+    function checkHandsDistance() {
+
+        var handPositionR = MyAvatar.getJointPosition('RightHand');
+        var handPositionL = MyAvatar.getJointPosition('LeftHand');
+        var handRotationR = MyAvatar.getRightPalmRotation();
+        var handRotationL = MyAvatar.getLeftPalmRotation();
+        var oldRotationR = previousHandOrientations[0];
+        var oldRotationL = previousHandOrientations[1];
+
+        if (Vec3.distance(handPositionL, handPositionR) <= handProximityDistance) {
+            if (!handsStretched) {
+                makeOpenPalm();
+                handsStretched = true;
+            }
+        } else {
+            if (handsStretched) {
+                clearJoints();
+                handsStretched = false;
+            }
+        }
+        if ((Vec3.distance(handPositionL, handPositionR) <= handProximityDistance) 
+            && (compareRotations(oldRotationR, handRotationR) 
+            || compareRotations(oldRotationL, handRotationL)) && applauseEnabled) {
+            playSingleClap();
+            previousHandOrientations = [];
+            previousHandOrientations.push(handRotationR);
+            previousHandOrientations.push(handRotationL);
+        }
+    }
+    // Tablet Handlers
+    function onClicked() {
+        if (!tablet) {
+            print("Warning in onClicked(): 'tablet' is undefined");
+            return;
+        }
+        if (open) {
+            tablet.gotoHomeScreen();
+        } else {
+            tablet.gotoWebScreen(appPage);
+        }
+    }
+    function onWebEventReceived(event) {
+        if (typeof event === 'string') {
+            event = JSON.parse(event);
+        }
+        if (!tablet) {
+            print("Warning in onWebEventReceived(): 'tablet' is undefined");
+            return;
+        }
+        switch (event.type) {
+            case 'applause-app-ready':
+                var webEvent = {
+                    type: 'setup',
+                    enabled : Settings.getValue(CAN_APPLAUD_SETTING),
```

Comment:
This one will return the expected value (false) if it's not explicitly set, but particles are set to true explictly because their default state if they don't already exist would otherwise be false.